logdb/logdb_test.go

```diff
@@ -428,3 +437,317 @@ func TestLogDB_HasBlockID(t *testing.T) {
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp directory: %w", err)
+	}
+
+	tmpFile, err := os.CreateTemp(dir, "temp-*.db")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp file: %w", err)
+	}
+
+	if err := tmpFile.Close(); err != nil {
+		return "", fmt.Errorf("failed to close temp file: %w", err)
+	}
+
+	return tmpFile.Name(), nil
+}
+
+func BenchmarkFakeDB_NewestBlockID(t *testing.B) {
+	dbPath, err := CreateTempDBPath()
+	defer os.Remove(dbPath)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	db, err := logdb.New(dbPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	b := new(block.Builder).Build()
+
+	b = new(block.Builder).
+		ParentID(b.Header().ID()).
+		Transaction(newTx()).
+		Build()
+	receipts := tx.Receipts{newReceipt()}
+
+	w := db.NewWriter()
+	if err := w.Write(b, receipts); err != nil {
+		t.Fatal(err)
+	}
+	if err := w.Commit(); err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name    string
+		prepare func() (thor.Bytes32, error)
+	}{
+		{
+			"newest block id",
+			func() (thor.Bytes32, error) {
+				return b.Header().ID(), nil
+			},
+		},
+		{
+			"add both event and transfer, best should change",
+			func() (thor.Bytes32, error) {
+				b = new(block.Builder).
+					ParentID(b.Header().ID()).
+					Transaction(newTx()).
+					Build()
+				receipts := tx.Receipts{newReceipt()}
+
+				w := db.NewWriter()
+				if err := w.Write(b, receipts); err != nil {
+					return thor.Bytes32{}, nil
+				}
+				if err := w.Commit(); err != nil {
+					return thor.Bytes32{}, nil
+				}
+				return b.Header().ID(), nil
+			},
+		},
+		{
+			"add event only, best should change",
+			func() (thor.Bytes32, error) {
+				b = new(block.Builder).
+					ParentID(b.Header().ID()).
+					Transaction(newTx()).
+					Build()
+				receipts := tx.Receipts{newEventOnlyReceipt()}
+
+				w := db.NewWriter()
+				if err := w.Write(b, receipts); err != nil {
+					return thor.Bytes32{}, nil
+				}
+				if err := w.Commit(); err != nil {
+					return thor.Bytes32{}, nil
+				}
+				return b.Header().ID(), nil
+			},
+		},
+		{
+			"add transfer only, best should change",
+			func() (thor.Bytes32, error) {
+				b = new(block.Builder).
+					ParentID(b.Header().ID()).
+					Transaction(newTx()).
+					Build()
+				receipts := tx.Receipts{newTransferOnlyReceipt()}
+
+				w := db.NewWriter()
+				if err := w.Write(b, receipts); err != nil {
+					return thor.Bytes32{}, nil
+				}
+				if err := w.Commit(); err != nil {
+					return thor.Bytes32{}, nil
+				}
+				return b.Header().ID(), nil
+			},
+		},
+	}
+
+	t.ResetTimer()
+	for _, tt := range tests {
+		t.Run(tt.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				want, err := tt.prepare()
+				if err != nil {
+					b.Fatal(err)
+				}
+				got, err := db.NewestBlockID()
+				if err != nil {
+					b.Fatal(err)
+				}
+				assert.Equal(b, want, got)
+			}
+		})
+	}
+}
+
+func BenchmarkFakeDB_WriteBlocks(t *testing.B) {
```

Comment:
For the Benchmark tests can a comment be added explaining what/how they're working ? Something clear so that we can easily pick back whenever we eventually come back to them